android/src/main/java/com/reactnativespokestack/Downloader.kt

```diff
@@ -0,0 +1,30 @@
+package com.reactnativespokestack
+
+import android.content.Context
+import android.util.Log
+import java.io.File
+import java.io.FileOutputStream
+import java.net.URL
+
+class Downloader(private val context: Context) {
+  private val logTag = "Spokestack"
+
+  fun downloadModel(filename: String, url: String): String {
+    val file = File(context.cacheDir, filename)
+    if (file.exists() && file.canRead() && file.length() > 0) {
```

Comment:
Returning an existing file unconditionally in the download function makes someone else responsible for deleting it if you have `refreshModels` enabled. That's fine, but it does also make that other party responsible for knowing the actual filesystem path the downloader's working with.

Reply:
I should add a way to clear the files.